src/sender-ui.py

```diff
@@ -21,70 +24,75 @@ def poll():
         sender.get_streams()
 
 
-def send_file():
-    global continue_sending
-    continue_sending = True
-    while continue_sending:
-        if sender.pending_streams:
-            print(
-                f"Pending Streams: {sender.pending_streams}, Completed Streams: {sender.completed_streams}"
-            )
-            stream_id = sender.pending_streams.pop(0)
-            ip, port = sender.consume_stream(stream_id)
-            if ip and port:
-                # To give time for receiver to start
-                # Need to find a more elegant solution in the future
-                time.sleep(3)
-                subprocess.Popen(
-                    [
-                        "ffmpeg",
-                        "-re",
-                        "-i",
-                        choose_file_entry.get(),
-                        "-f",
-                        "mpegts",
-                        "-v",
-                        "warning",
-                        f"srt://{ip}:{port}?pkt_size=1316",
-                    ]
-                )
-            sender.completed_streams.append(stream_id)
-
-
-def send_cam():
+def send(use_webcam: bool):
     global continue_sending
     continue_sending = True
     with open("config.json") as json_config:
         config = json.load(json_config)
     webcam = config["camera"]["name"]
     while continue_sending:
         if sender.pending_streams:
-            print(
-                f"Pending Streams: {sender.pending_streams}, Completed Streams: {sender.completed_streams}"
-            )
+            print(f"Pending Streams: {sender.pending_streams}, Completed Streams: {sender.completed_streams}")
+
             stream_id = sender.pending_streams.pop(0)
-            ip, port = sender.consume_stream(stream_id)
+            ip, port, is_rendezvous = sender.consume_stream(stream_id)
             if ip and port:
                 # To give time for receiver to start
                 # Need to find a more elegant solution in the future
                 time.sleep(3)
+                if use_webcam:
+                    start_ffmpeg_webcam(webcam)
+                else:
+                    start_ffmpeg_file()
+
+                if is_rendezvous:
+                    srt_url = f"{SRT_SCHEME}://{ip}:{port}?mode=rendezvous"
+                else:
+                    srt_url = f"{SRT_SCHEME}://{ip}:{port}"
```

Comment:
Won't work if both samples are local (i.e same public ip). So in the future we'll have to check for that case (but maybe we don't even want to? Not sure)

Also, how does the isRendezVous field work in the service? Ideally it should be automatically set to true when the stream involves devices that are on separate networks (i.e different public ips)

Reply:
> Also, how does the isRendezVous field work in the service? Ideally it should be automatically set to true when the stream involves devices that are on separate networks (i.e different public ips)

That's what the backend is doing right now, although it hasn't been thoroughly tested yet, because it is so complicated to test. Here's a code snippet from `StreamServiceImp` that makes the rendezvous mode decisions.
```Java
private boolean shouldUseRendezvousMode(
    InputChannelDto inputChannelDto, OutputChannelDto outputChannelDto) {
  DeviceDto decoderDevice = inputChannelDto.getDecoder().getDevice();
  DeviceDto encoderDevice = outputChannelDto.getEncoder().getDevice();
  return !(networkingUtil.areDevicesOnSameLocalNetworkAsService(decoderDevice, encoderDevice)
      || networkingUtil.areDevicesOnSamePrivateNetwork(decoderDevice, encoderDevice));
}
```